sycl/test-e2e/AOT/fpga-aoc-archive-early2.cpp

```diff
@@ -0,0 +1,35 @@
+// Test compiler behaviors for -fintelfpga with -fsycl-link=early.
```

Comment:
2 in the filename isn't much meaningful. Rename the file to sycl/test-e2e/AOT/fpga-aoc-archive-early-image.cpp?

Reply:
There are three existing test files before the splitting.
| Test file     | Test purpose |
| ------------- | ------------- |
| fpga-aoc-archive.cpp | mixed usage of early and image archives |
| fpga-aoc-archive-early.cpp | usage of early archive  |
| fpga-aoc-archive-image.cpp | usage of image archive|

File names `fpga-aoc-archive2.cpp` and `fpga-aoc-archive-early2.cpp` should indicate the test purpose more clearly than names like `fpga-aoc-archive-early-image.cpp`.

Reply:
LGTM